### PR TITLE
program: remove unused tiny_bip39 dep

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
This dep is unused

#### Summary of Changes
Remove it

### CODE REVIEW

**Comment by kevinheavey** [comment]
If these wasm test failures have anything to do with the changes then I can't tell why.

**Comment by kevinheavey** [comment]
Figured it out, tiny_bip39 was the only thing activating the wasm-bindgen feature of rand, which we need. I've now added the feature explicitly:
```
[target.'cfg(target_arch = "wasm32")'.dependencies]
...
rand = { version = "0.7.3", features = ["wasm-bindgen"] }
```

File: sdk/program/Cargo.toml
Comment (buffalojoec) [inline sdk/program/Cargo.toml:75]: `version = workspace`? Also, does this need to be added if `tiny-bip39` comes out?

**Comment by buffalojoec** [comment]
> Figured it out, tiny_bip39 was the only thing activating the wasm-bindgen feature of rand, which we need. I've now added the feature explicitly:
> 
> ```
> [target.'cfg(target_arch = "wasm32")'.dependencies]
> ...
> rand = { version = "0.7.3", features = ["wasm-bindgen"] }
> ```

Can we use the workspace version?

**Comment by kevinheavey** [comment]
> > Figured it out, tiny_bip39 was the only thing activating the wasm-bindgen feature of rand, which we need. I've now added the feature explicitly:
> > ```
> > [target.'cfg(target_arch = "wasm32")'.dependencies]
> > ...
> > rand = { version = "0.7.3", features = ["wasm-bindgen"] }
> > ```
> 
> Can we use the workspace version?

No, it doesn't compile due to version constraints

**Comment by buffalojoec** [comment]
Mind rebasing this one as well? It should pull in the flaky CI fix.

Hm, I really wish this could use the workspace version. It makes it tricky to update dependencies this way.

**Comment by kevinheavey** [comment]
Looks like failing tests just need to retry:
```
[2024-04-30T23:14:14.925128473Z ERROR cargo_build_sbf] Failed to install platform-tools: error sending request for url (https://github.com/solana-labs/platform-tools/releases/download/v1.41/platform-tools-linux-x86_64.tar.bz2): error trying to connect: Connection reset by peer (os error 104)
```

**Comment by buffalojoec** [comment]
Sorry to leave you hanging here @kevinheavey, but this works for me. I see a few other crates using non-workspace versions, including ZK libs which use `rand` 0.7. Also it's only a dependency for `wasm` builds.

But I will defer to @joncinque or @CriesofCarrots for the stamp, in case we've been trying to avoid this.

**Review by joncinque — COMMENTED** [review]
The change looks good, but I tried on my local machine and my dev box, and I can't get the error to trigger if I remove the extra `rand` line, and I can't find the previous CI runs to see what went wrong unfortunately. Was it a particular configuration?

I'm surprised you were seeing the issue, since the wasm build pulls the rand version from the `not(target_os = "solana")` part, and then `getrandom` is correctly enabling the `js` feature. The `wasm-bindgen` flag doesn't exist on rand v0.8, so we *should* be doing this correctly.

Sorry for the back-and-forth, but could you try removing the rand line? Or give some more info about the test failure? If we remove that line, we'll also have one less crate to compile :pray:

**Comment by joncinque** [comment]
Ohhhh the problem is in `solana-sdk`, not `solana-program`. Ah darn, let me take a look around and see what solana-sdk needs

**Comment by joncinque** [comment]
Ok this was painful, but in `solana-sdk`'s Cargo.toml, we need to add:

```
getrandom = { version = "0.1", features = ["wasm-bindgen"] }
```

Under the `[target.'cfg(target_arch = "wasm32")'.dependencies]` section. This is because we need to enable the `wasm-bindgen` feature on ed25519-dalek's dependency of getrandom, which is still on 0.1.

If you're ok with it, I can push the change, or you can do it, all fine by me!

**Comment by kevinheavey** [comment]
About to go sleep so fine by me if you do it!

**Review by joncinque — APPROVED** [review]
Thanks for the contribution!

**Comment by joncinque** [comment]
Ah never mind, my approval doesn't count since I pushed the last commit, can I get an approval @buffalojoec ?

**Review by buffalojoec — APPROVED** [review]

**Pull request merged**
